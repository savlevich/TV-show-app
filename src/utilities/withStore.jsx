import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const withStore = (Component, initialState = {}) => () => {
  const mockStore = configureStore();
  const store = mockStore(initialState);
  return <Provider store={store}>{Component}</Provider>;
};

export default withStore;
