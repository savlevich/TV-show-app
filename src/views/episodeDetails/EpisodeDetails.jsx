import React, { useEffect } from "react";
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchEpisodeData, clearCurrentEpisodeData } from './ducks/actions';
import {
  getEpisodeNameSelector,
  getEpisodeDescriptionSelector, 
  getEpisodePictureSelector,
} from './ducks/selectors';
import ContentWrapper from '../../components/ContentWrapper';
import './styles.scss';
import NoImage from '../../images/no-image.png';

const EpisodeDetails = () => {
  const { episodeId } = useParams();

  const dispatch = useDispatch();
  const episodeName = useSelector(getEpisodeNameSelector);
  const pictureUrl = useSelector(getEpisodePictureSelector);
  const episodeDescription = useSelector(getEpisodeDescriptionSelector);

  useEffect(() => {
    const fetchData = async episodeId => {
      dispatch(fetchEpisodeData(episodeId));
    }
    if (episodeId) {
      fetchData(episodeId);
    }

    return () => {
      dispatch(clearCurrentEpisodeData());
    }
  }, [dispatch, episodeId]);

  return (
    <ContentWrapper className="episode">
    <div className="episode__info">
      <h2 className="episode__title">{episodeName}</h2>
      <div className="episode__picture__wrapper">
        <img width="300" height="400" className="episode__picture" alt="show" src={pictureUrl || NoImage} />
      </div>
      <div className="episode__desc" dangerouslySetInnerHTML={{ __html: episodeDescription }} />
    </div>
    </ContentWrapper>
  )
};

export default EpisodeDetails;