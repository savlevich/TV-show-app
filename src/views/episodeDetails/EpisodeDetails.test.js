import React from 'react';
import { render } from '@testing-library/react';
import { useParams, MemoryRouter } from 'react-router-dom';
import withStore from '../../utilities/withStore';
import EpisodeDetails from './EpisodeDetails';

const store = {
  currentEpisode: {
    episodeDetails: {
      name: 'The Wrinklegruff Gals'
    },
  },
};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
}));

describe('EpisodeDetails', () => {
  test('should render EpisodeDetails', () => {
    useParams.mockReturnValue({ episodeId: 'test' });

    const ElementWithStore = withStore(
      <MemoryRouter>
        <EpisodeDetails />
      </MemoryRouter>, store);
    const { asFragment } = render(<ElementWithStore />);

    expect(asFragment()).toMatchSnapshot();
  });
});
