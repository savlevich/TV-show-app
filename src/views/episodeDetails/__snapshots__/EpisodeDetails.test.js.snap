// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeDetails should render EpisodeDetails 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper"
  >
    <div
      class="episode__info"
    >
      <h2
        class="episode__title"
      >
        The Wrinklegruff Gals
      </h2>
      <div
        class="episode__picture__wrapper"
      >
        <img
          alt="show"
          class="episode__picture"
          height="400"
          src="no-image.png"
          width="300"
        />
      </div>
      <div
        class="episode__desc"
      />
    </div>
  </div>
</DocumentFragment>
`;
